language/tools/test_generation/src/control_flow_graph.rs

```diff
@@ -144,25 +157,21 @@ impl CFG {
             !block_ids.is_empty(),
             "Cannot merge locals of empty block list"
         );
+        let first_basic_block = self.basic_blocks.get(&block_ids[0]);
+        // Implication of preconditon
+        assume!(first_basic_block.is_some());
+        let first_basic_block_locals_out = &first_basic_block.unwrap().locals_out;
+        let locals_len = first_basic_block_locals_out.len();
         let mut locals_out = BlockLocals::new();
-        let locals_len = self
-            .basic_blocks
-            .get(&block_ids[0])
-            .unwrap()
-            .locals_out
-            .len();
         for local_index in 0..locals_len {
-            let abstract_value = self.basic_blocks.get(&block_ids[0]).unwrap().locals_out
-                [&local_index]
-                .0
-                .clone();
+            let abstract_value = first_basic_block_locals_out[&local_index].0.clone();
             let mut availability = BorrowState::Available;
             for block_id in block_ids.iter() {
                 // A local is available for a block if it is available in every
                 // parent's outgoing locals
-                if self.basic_blocks.get(block_id).unwrap().locals_out[&local_index].1
-                    == BorrowState::Unavailable
-                {
+                let basic_block = self.basic_blocks.get(block_id);
+                assume!(basic_block.is_some());
```

Comment:
Every assume should be justified by a comment.

Reply:
Added a comment in the next PR.